install_sct

```diff
@@ -599,13 +599,11 @@ python/bin/conda create -y -n venv_sct python=3.7
 find "$SCT_DIR/$PYTHON_DIR" -type f -exec touch {} +
 
 # make sure that there is no conflict with local python install by making venv_sct an isolated environment.
-# workaround for https://github.com/conda/conda/issues/7173
+# thanks @javidcf https://github.com/conda/conda/issues/7173#issuecomment-884098367
```

Comment:
To test this PR, I installed `python 3.7.12` and set it as my default `python` outside of `venv_sct`:

```console
joshua@tadpole:~$ pyenv install 3.7.12
joshua@tadpole:~$ pyenv global 3.7.12
```

Next, I installed a conflicting version of `numpy` to the user site packages that corresponded to my default Python:

```console
joshua@tadpole:~$ pip --version
pip 20.1.1 from /home/joshua/.pyenv/versions/3.7.12/lib/python3.7/site-packages/pip (python 3.7)
joshua@tadpole:~$ pip install --user numpy==1.19.0rc1 
```

Finally, I compared `sct_check_dependencies` under a variety of conditions:

<details>
<summary> With `PYTHONNOUSERSITE` set AND venv_sct active </summary>

```console
(venv_sct) joshua@tadpole:~/repos/spinalcordtoolbox$ echo $PYTHONNOUSERSITE
1
(venv_sct) joshua@tadpole:~/repos/spinalcordtoolbox$ sct_check_dependencies

--
Spinal Cord Toolbox (git-ng/simpler-isolated-conda-caaf5b200e71db431a36d44f3afcf2d92d68cfd6)

sct_check_dependencies 
--

SCT info:
- version: git-ng/simpler-isolated-conda-caaf5b200e71db431a36d44f3afcf2d92d68cfd6
- path: /home/joshua/repos/spinalcordtoolbox
OS: linux (Linux-5.11.0-40-generic-x86_64-with-debian-bullseye-sid)
CPU cores: Available: 8, Used by ITK functions: 8
RAM: Total: 15913MB, Used: 5788MB, Available: 9729MB
Check Python executable.............................[OK]
  Using bundled python 3.7.11 (default, Jul 27 2021, 14:32:16) 
[GCC 7.5.0] at /home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.3)
Check if dipy is installed..........................[OK] (1.4.1)
Check if h5py is installed..........................[OK] (2.10.0)
Check if Keras (2.3.1) is installed.................[OK] (2.3.1)
Check if ivadomed is installed......................[OK] (2.7.4)
Check if matplotlib is installed....................[OK] (3.5.0)
Check if nibabel is installed.......................[OK] (3.2.1)
Check if numpy is installed.........................[OK] (1.18.5)
Check if onnxruntime (1.4.0) is installed...........[OK] (1.4.0)
Check if pandas is installed........................[OK] (1.3.4)
Check if psutil is installed........................[OK] (5.8.0)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.2.5)
Check if pytest-cov is installed....................[OK] (3.0.0)
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.26.0)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.7.3)
Check if scikit-image is installed..................[OK] (0.18.3)
Check if scikit-learn is installed..................[OK] (1.0.1)
Check if tensorflow is installed....................[OK] (1.15.5)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.62.3)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.26.7)
Check if pytest_console_scripts is installed........[OK]
Check if wquantiles is installed....................[OK] (0.4)
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if figure can be opened with matplotlib.......[OK] (Using GUI backend: 'QtAgg')
Check if figure can be opened with PyQt.............[OK]
Check FSLeyes version...............................[OK] (1.3.0)
```
</details>

<details>
<summary> With `PYTHONNOUSERSITE` set, NO venv_sct active </summary>

```console

sct_check_dependencies 
--

SCT info:
- version: git-ng/simpler-isolated-conda-caaf5b200e71db431a36d44f3afcf2d92d68cfd6
- path: /home/joshua/repos/spinalcordtoolbox
OS: linux (Linux-5.11.0-40-generic-x86_64-with-debian-bullseye-sid)
CPU cores: Available: 8, Used by ITK functions: 8
RAM: Total: 15913MB, Used: 5779MB, Available: 9738MB
Check Python executable.............................[OK]
  Using bundled python 3.7.11 (default, Jul 27 2021, 14:32:16) 
[GCC 7.5.0] at /home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.3)
Check if dipy is installed..........................[OK] (1.4.1)
Check if h5py is installed..........................[OK] (2.10.0)
Check if Keras (2.3.1) is installed.................[OK] (2.3.1)
Check if ivadomed is installed......................[OK] (2.7.4)
Check if matplotlib is installed....................[OK] (3.5.0)
Check if nibabel is installed.......................[OK] (3.2.1)
Check if numpy is installed.........................[OK] (1.19.0rc1)
Check if onnxruntime (1.4.0) is installed...........[OK] (1.4.0)
Check if pandas is installed........................[OK] (1.3.4)
Check if psutil is installed........................[OK] (5.8.0)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.2.5)
Check if pytest-cov is installed....................[OK] (3.0.0)
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.26.0)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.7.3)
Check if scikit-image is installed..................[OK] (0.18.3)
Check if scikit-learn is installed..................[OK] (1.0.1)
Check if tensorflow is installed....................[OK] (1.15.5)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.62.3)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.26.7)
Check if pytest_console_scripts is installed........[OK]
Check if wquantiles is installed....................[OK] (0.4)
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if figure can be opened with matplotlib.......[OK] (Using GUI backend: 'QtAgg')
Check if figure can be opened with PyQt.............[OK]
Check FSLeyes version...............................Exit code 120 occurred. Please report this issue on SCT GitHub: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues
```
</details>

So, it looks like this change only works if `venv_sct` is explicitly activated. But, that's not how users typically use SCT, since we modify the path to make our executables available _without_ needing to activate `venv_sct`. :sweat:

Reply:
Dangit

Thanks for doing all that testing then!!

Reply:
What about https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3606#issuecomment-980496770 ?

Reply:
Sounds peachy-keen. :smile: 

The only thing I noticed is that [your comment](https://github.com/conda/conda/issues/7173#issuecomment-980496682) mentions `python3.6/site.py`. Thankfully, the [Python 3.10](https://docs.python.org/3/library/site.html#module-site) docs preserve the bit about `pyvenv.cfg`, so I imagine the fix will continue to work moving forward, too. 

I'll try it out just to be sure!

Reply:
It worked, but I had to make a very slight addendum to the suggestion:

```diff
-echo "include-system-site-packages = false" > $SCT_DIR/$PYTHON_DIR/pyvenv.cfg
+echo "include-system-site-packages = false" > $SCT_DIR/$PYTHON_DIR/envs/venv_sct/pyvenv.cfg
```

Reply:
Oh yes, of course. Great!